doc/src/tutorial/assumptions.rst

```diff
@@ -0,0 +1,1836 @@
+
+The values given in an assumptions query use three-valued "fuzzy" logic. Any
+query can return ``True``, ``False``, or ``None`` where ``None`` should be
+interpreted as meaning that the result is *unknown*.
+
+    >>> x = Symbol('x')
+    >>> y = Symbol('y', positive=True)
+    >>> z = Symbol('z', negative=True)
+    >>> print(x.is_positive)
+    None
+    >>> print(y.is_positive)
+    True
+    >>> print(z.is_positive)
+    False
+
+.. note:: We need to use ``print`` in the above examples because the special
+          value ``None`` does not display by default in the Python
+          interpretter.
+
+There are two reasons why an assumptions query might give ``None``. It is
+possible that the query is *unknowable* as in the case of ``x`` above. Since
+``x`` does not have any assumptions declared it roughly represents an
+arbitrary complex number. An arbitrary complex number *might* be a positive
+real number but it also might *not* be. Without further information there is
+no way to resolve the query ``x.is_positive``.
+
+The other reason that an assumptions query might give ``None`` is just that
+the assumptions system does not try very hard to answer complicated queries.
+The system is intended to be fast and uses simple heuristic methods to
+conclude a ``True`` or ``False`` answer in common cases. For example any sum
+of positive terms is positive so:
+
+    >>> from sympy import symbols
+    >>> x, y = symbols('x, y', positive=True)
+    >>> expr = x + y
+    >>> expr
+    x + y
+    >>> expr.is_positive
+    True
+
+The last example is particularly simple so the assumptions system is able to
+give a definite answer. If the sum involved a mix of positive or negative
+terms it would be a harder query:
+
+    >>> x = Symbol('x', real=True)
+    >>> expr = 1 + (x - 2)**2
+    >>> expr
+    (x - 2)**2 + 1
+    >>> expr.is_positive
+    True
+    >>> expr2 = expr.expand()
+    >>> expr2
+    x**2 - 4*x + 5
+    >>> print(expr2.is_positive)
+    None
+
+Ideally that last example would give ``True`` rather than ``None`` because the
+expression is always positive for any real value of ``x`` (and ``x`` has been
+assumed real). The assumptions system is intended to be efficient though: it
+is expected many more complex queries will not be fully resolved. This is
+because assumptions queries are primarily used internally by SymPy as part of
+low-level calculations. Making the system more comprehensive would slow SymPy
+down.
+
+Note that in fuzzy logic giving an indeterminate result ``None`` is never a
+contradiction. If it is possible to infer a definite ``True`` or ``False``
+result when resolving a query then that is better than returning ``None``.
+However a result of ``None`` is not a *bug*. Any code that uses the
+assumptions system needs to be prepared to handle all three cases for any
+query and should not presume that a definite answer will always be given.
+
+The assumptions system is not just for symbols or for complex expressions. It
```

Comment:
I would remove this last paragraph and example. We already talk about using `S` earlier in the tutorial, and the details about `Basic` are implementation details that don't need to be in the tutorial.